From 57681cf0162865bd6f418fa1f9477b73a896e414 Mon Sep 17 00:00:00 2001
From: Jani Hautakangas <jani@igalia.com>
Date: Sun, 22 May 2022 23:03:03 +0300
Subject: [PATCH 5/7] Workaround for Android emulator samplerExternalOES
 precision issue

For some unknown reason glShaderSource fails with 0x505 GL_OUT_OF_MEMORY
on Android Emulator if precision is not set right after SamplerExternalOESType
definition. Seems like emulator OpenGL ES implementation doesn't define any
precision for samplerExternalOES. Adding precision hint for samplerExternalOES
across all platforms is harmless because OpenGL ES implementation is not
required to use hint information.
---
 .../platform/graphics/texmap/TextureMapperShaderProgram.cpp      | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp b/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
index 58863926b5fe..4a7b656bd729 100644
--- a/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
+++ b/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
@@ -170,6 +170,7 @@ static const char* vertexTemplateCommon =
     GLSL_DIRECTIVE(ifdef ENABLE_TextureExternalOES) \
         GLSL_DIRECTIVE(extension GL_OES_EGL_image_external : require) \
         GLSL_DIRECTIVE(define SamplerExternalOESType samplerExternalOES) \
+        STRINGIFY(precision mediump samplerExternalOES;\n) \
     GLSL_DIRECTIVE(else) \
         GLSL_DIRECTIVE(define SamplerExternalOESType sampler2D) \
     GLSL_DIRECTIVE(endif)
-- 
2.34.1

